pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -136,3 +159,109 @@ func (e shortcutExpander) expandResourceShortcut(resource schema.GroupVersionRes
 
 	return resource
 }
+
+func (e *shortcutExpander) wrapError(err error, resource schema.GroupVersionResource) error {
+	switch er := err.(type) {
+	case *meta.NoResourceMatchError:
+		return &NoResourceWithCommendError{
+			NoResourceMatchError: er,
+			Recommends:           e.similarResources(resource),
+		}
+	}
+
+	return err
+}
+
+func (e *shortcutExpander) similarResources(resource schema.GroupVersionResource) []string {
```

Comment:
this is separate function... make it a separate component that takes either a discovery interface or a list of resources, along with the specified resource, and makes suggestions